crates/primitives/src/signature/sig.rs

```diff
@@ -126,47 +114,35 @@ impl Signature<k256::ecdsa::Signature> {
         .unwrap()
     }
 
+    /// Instantiate a new signature from `r`, `s`, and `v` values.
+    pub const fn new(r: U256, s: U256, v: Parity) -> Self {
+        Self { r, s, v }
+    }
+
     /// Instantiate from a signature and recovery id
+    #[cfg(feature = "k256")]
     pub fn from_signature_and_parity<T: TryInto<Parity, Error = E>, E: Into<SignatureError>>(
         sig: k256::ecdsa::Signature,
         parity: T,
     ) -> Result<Self, SignatureError> {
         let r = U256::from_be_slice(sig.r().to_bytes().as_ref());
         let s = U256::from_be_slice(sig.s().to_bytes().as_ref());
-        Ok(Self { inner: sig, v: parity.try_into().map_err(Into::into)?, r, s })
-    }
-
-    /// Instantiate from v, r, s.
-    pub fn from_rs_and_parity<T: TryInto<Parity, Error = E>, E: Into<SignatureError>>(
-        r: U256,
-        s: U256,
-        parity: T,
-    ) -> Result<Self, SignatureError> {
-        Self::from_scalars_and_parity(r.into(), s.into(), parity)
-    }
-
-    /// Parses a signature from a byte slice, with a v value
-    #[inline]
-    pub fn from_bytes_and_parity<T: TryInto<Parity, Error = E>, E: Into<SignatureError>>(
```

Comment:
why are we not keeping these?

Reply:
we are, they were duplicated in two impl blocks

https://github.com/alloy-rs/core/pull/719/files#diff-79b82cfeeba5106f10ab06fb251c628206b2d8f648bcd94e6fe01c6091763d1cR239-R249